src/mmw/js/src/core/models.js

```diff
@@ -61,36 +61,89 @@ var TaskModel = Backbone.Model.extend({
         }
     },
 
+    // Cancels any currently running jobs. The promise returned
+    // by previous calls to pollForResults will be rejected.
+    reset: function() {
+        this.set({
+            'job': null,
+            'result': null,
+            'status': null
+        });
+    },
+
+    // taskHelper should be an object containing an optional object,
+    // postData, an optional function, onStart, and functions pollSuccess,
+    // pollFailure, and startFailure.
+    start: function(taskHelper) {
+        this.reset();
+        if (taskHelper.onStart) {
+            taskHelper.onStart();
+        }
+        var self = this;
+
+        self
+            .fetch({
+                method: 'POST',
+                data: taskHelper.postData
+            })
+            .done(function() {
+                self.pollForResults()
+                    .done(function() {
+                        taskHelper.pollSuccess();
+                    })
+                    .fail(function(failure) {
+                        if (failure && failure.reset) {
+                            console.log('Job ' + failure.reset.job + ' was cancelled.');
+                        } else {
+                            taskHelper.pollFailure();
+                        }
+                    });
+            })
+            .fail(function() {
+                taskHelper.startFailure();
+            });
+    },
+
     pollForResults: function() {
         var defer = $.Deferred(),
             duration = 0,
-            self = this;
+            self = this,
+            expectedJob = self.get('job');
 
         // Check the task endpoint to see if the job is
         // completed. If it is, return the results of
         // the job. If not, check again after
         // pollInterval has elapsed.
         var getResults = function() {
-                if (duration >= self.get('timeout')) {
-                    defer.reject();
-                    return;
-                }
+            if (duration >= self.get('timeout')) {
+                defer.reject();
+                return;
+            }
 
-                self.fetch()
-                    .done(function(response) {
-                        console.log('Polling ' + self.url());
-                        if (response.status !== 'complete') {
-                            duration = duration + self.get('pollInterval');
-                            window.setTimeout(getResults, self.get('pollInterval'));
-                        } else {
-                            defer.resolve(response);
-                        }
-                    })
-                    .fail(defer.reject);
+            // If job was cancelled.
+            if (expectedJob != self.get('job')) {
+                defer.reject({
+                    reset: {
```

Comment:
Consider flattening this data structure, if possible. IMO `reject({ reset: { job: ... } })` isn't more descriptive than `reject(expectedJob)` or maybe even `reject(expectedJob + ' was cancelled')`.